### ISSUE TITLE
TAFs with missing issue date/time

### ISSUE THREAD

**Issue by MichaelRoentgen** [dmbaturin/pytaf#16]
I noticed a few TAFs in my dataset that had no issue date/time and that the parser then incorrectly interpreted the valid form/to date/time as issue date/time, e. g.:

'TAF ENBO 0615/0715 03008KT 9999 FEW025 SCT100 BECMG 0615/0618 25010KT PROB30 TEMPO 0621/0708 BKN008'

would be interpreted as

TAF for KQHN **issued 07: UTC** on the 02nd, **valid from :00 UTC on the  to** 13:00 UTC on the 03rd
    Wind: from 270 degrees at 09 knots 
    Visibility: more than 10 000 meters 
    Sky conditions: sky clear 
 
Temporarily between 11:00 on the 02nd and 15:00 on the 02nd: 
    Wind: from 270 degrees at 10 knots gusting to 15 knots 
    Visibility: 6000 meters 
    Weather: blowing dust 
 
Gradual change to the following between 12:00 on the 02nd and 13:00 on the 02nd: 
    Wind: variable at 06 knots 
    Visibility: 9000 meters 
    Sky conditions: sky clear 
    Weather: haze 

I am not familiar enough with regex to come up with a sensible solution to this? Is there a way to define that the "valid from/to" expression must be encapsulated by spaces and must contain a /?

Kind regards
Mike

**Comment by dmbaturin** [comment]
Interesting question. Frankly, I'm tempted to rewrite the parser part as a custom-built state machine, or using some context-sensitive language toolkit. The "regex"-based approach obviously got out of control and reached its maintainability limits quickly.

**Comment by jamayfieldjr** [comment]
I don't believe so dmbaturin regex is really powerful and you've obviously built a program that works! I've have a program in MATLAB that decodes TAFs. There are only a number of combination the NWS can put in their TAFs. Eurocontrol in Brussels is a bit more lean and use a ton of prob groups! IT DRIVES ME CRAZY! I found your program a year ago and have really gotten into python3.
